tools/c7n_gcp/c7n_gcp/resources/spanner.py

```diff
@@ -35,10 +36,108 @@ def get(client, resource_info):
             )
 
 
+@SpannerInstance.action_registry.register('delete')
+class SpannerInstanceDelete(MethodAction):
+    """The action is used for spanner instances delete.
+    GCP action is https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances/delete
+
+    Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-spanner-instances-delete
+            resource: gcp.spanner-instance
+            filters:
+              - type: value
+                key: nodeCount
+                op: gte
+                value: 2
+            actions:
+              - type: delete
+    """
+    schema = type_schema('delete')
+    method_spec = {'op': 'delete'}
+
+    def get_resource_params(self, model, resource):
+        return {'name': resource['name']}
+
+
+@SpannerInstance.action_registry.register('set')
+class SpannerInstancePatch(MethodAction):
+    """The action is used for spanner instances nodeCount patch.
+    GCP action is https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances/patch
+
+    Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-spanner-instances-change-node-count
+            resource: gcp.spanner-instance
+            filters:
+              - type: value
+                key: nodeCount
+                op: gte
+                value: 2
+            actions:
+              - type: set
+                nodeCount: 1
+    """
+    schema = type_schema('set', required=['nodeCount'],
+                         **{'nodeCount': {'type': 'number'}})
+    method_spec = {'op': 'patch'}
+
+    def get_resource_params(self, model, resource):
+        result = {'name': resource['name'],
+                  'body': {
+                      'instance': {
+                          'nodeCount': self.data['nodeCount']
+                      },
+                      'field_mask': ', '.join(['nodeCount'])}
+                  }
+        return result
+
+
+@SpannerInstance.action_registry.register('set-iam-policy')
+class SpannerInstanceSetIamPolicy(SetIamPolicyBaseAction):
+    """Sets IAM policy. It works with bindings only.
+    GCP action is
+    https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances/setIamPolicy.
+
+    There are available following member types:
+    - allUsers,
+    - allAuthenticatedUsers,
+    - user,
+    - group,
+    - domain,
+    - serviceAccount.
+
+    Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-spanner-instances-set-iam-policy
+            resource: gcp.spanner-instance
+            actions:
+              - type: set-iam-policy
+                bindings:
+                  - members:
+                      - user:user1@test.com
+                      - user:user2@test.com
+                    role: roles/owner
+                  - members:
+                      - user:user3@gmail.com
+                    role: roles/viewer
+    """
+    pass
```

Comment:
remove the pass, rather than subclass for the sake of a doc string, do the doc string on the base class and just register it for use .. ie it becomes a common action in that case.

Reply:
I have updated this part. The doc string was moved into base class. The link to GCP action was removed. Please check if it's good implementation. The pass was left due to absence the doc string in the class

Reply:
don't need the subclass to register an action just do

```
SpannerInstance.action_registry.register('set-iam-policy', SetIamPolicy)
```
also rename the base class to reflect its direct usage.

Reply:
Updated